utils/autotailor.py

```diff
@@ -0,0 +1,193 @@
+        self.profile_title = ""
+
+        self.value_changes = []
+        self.rules_to_select = []
+        self.rules_to_unselect = []
+
+    def _full_id(self, string, default_prefix):
+        if id_is_full(string):
+            return string
+        return default_prefix + string
+
+    def _full_profile_id(self, string):
+        return self._full_id(string, self.PROFILES_PREFIX)
+
+    def _full_var_id(self, string):
+        return self._full_id(string, self.VARS_PREFIX)
+
+    def _full_rule_id(self, string):
+        return self._full_id(string, self.RULES_PREFIX)
+
+    def add_value_change(self, varname, value):
+        self.value_changes.append((varname, value))
+
+    def to_xml(self, location=None):
+        root = ET.Element("xccdf-1.2:Tailoring")
+        root.set("xmlns:xccdf-1.2", ALL_NS["xccdf-1.2"])
+        root.set("id", "xccdf_" + self.id_suffix)
+
+        benchmark = ET.SubElement(root, "xccdf-1.2:benchmark")
+        benchmark.set("href", self.original_ds_filename)
+
+        version = ET.SubElement(root, "xccdf-1.2:version")
+        version.set("time", datetime.datetime.now().isoformat())
+        version.text = str(self.version)
+
+        profile = ET.SubElement(root, "xccdf-1.2:Profile")
+        profile.set("id", self._full_profile_id(self.profile_name))
+        profile.set("extends", self._full_profile_id(self.extends))
+
+        # Title has to be there due to the schema definition.
+        title = ET.SubElement(profile, "xccdf-1.2:title")
+        if self.profile_title:
+            title.set("override", "true")
+        else:
+            title.set("override", "false")
+        title.text = self.profile_title
+
+        for rule_id in self.rules_to_select:
+            change = ET.SubElement(profile, "xccdf-1.2:select")
+            change.set("idref", self._full_rule_id(rule_id))
+            change.set("selected", "true")
+
+        for rule_id in self.rules_to_unselect:
+            change = ET.SubElement(profile, "xccdf-1.2:select")
+            change.set("idref", self._full_rule_id(rule_id))
+            change.set("selected", "false")
+
+        for varname, value in self.value_changes:
+            change = ET.SubElement(profile, "xccdf-1.2:set-value")
+            change.set("idref", self._full_var_id(varname))
+            change.text = str(value)
+
+        if location:
+            ET.ElementTree(root).write(location)
+        else:
+            try:
+                # Python >= 3.9
+                ET.indent(root)
+            except AttributeError:
+                ET.dump(root)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="This script produces XCCDF 1.2 tailoring files to be used by SCAP scanners and SCAP datastreams.")
+    parser.add_argument(
+        "datastream", metavar="DS-FILENAME",
+        help="The datastream filename is just informational, "
+        "it doesn't have any key role in the tailoring composition.")
```

Comment:
If it's just informational and not key role, then why is it a required argument?

Reply:
It is required, because you need to put some kind of datastream href into the tailoring. I know that it is silly, but the workbench does it the same way.

> ... The 1<xccdf:benchmark>1 element holds the associated Benchmark's **location URI**, version string, and id value. ...

Reply:
That was actually my guess. Then, please change the help text.